test/graphql/schemas/args/argGenerator.test.ts

```diff
@@ -0,0 +1,186 @@
+      instance.id = { eq: "123" };
+      instance.name = { contains: "test" };
+      expect(instance.id).toEqual({ eq: "123" });
+      expect(instance.name).toEqual({ contains: "test" });
+    });
+
+    it("should handle referenced entities", () => {
+      const { WhereArgs } = createEntityArgs<TestEntity>("Test", {
+        id: "id",
+        name: "string",
+        reference: {
+          type: "id",
+          references: {
+            entity: ReferencedEntity,
+            fields: {
+              count: "number",
+            },
+          },
+        },
+      });
+
+      const instance = new WhereArgs();
+      expect(Object.keys(instance)).toContain("reference");
+
+      instance.reference = { count: { gt: 5 } };
+      expect(instance.reference).toEqual({ count: { gt: 5 } });
+    });
+
+    it("should handle partial field definitions", () => {
+      const { WhereArgs } = createEntityArgs<TestEntity>("Test", {
+        id: "id", // Only defining id, omitting other fields
+      });
+
+      const instance = new WhereArgs();
+
+      expect(Object.keys(instance)).toEqual(["id"]);
+
+      instance.id = { eq: "123" };
+      expect(instance.id).toEqual({ eq: "123" });
+
+      expect(instance.name).toBeUndefined();
+      expect(instance.reference).toBeUndefined();
+    });
+  });
+
+  describe("SortArgs", () => {
+    it("should create sort args with correct fields", () => {
+      const { SortArgs } = createEntityArgs<TestEntity>("Test", {
+        id: "id",
+        name: "string",
+      });
+
+      const instance = new SortArgs();
+
+      // Test valid sort orders
+      instance.by = {
+        id: SortOrder.ascending,
+        name: SortOrder.descending,
+      };
+      expect(instance.by).toEqual({
+        id: SortOrder.ascending,
+        name: SortOrder.descending,
+      });
+    });
+
+    it("should default to ascending for invalid sort orders", () => {
+      const { SortArgs } = createEntityArgs<TestEntity>("Test", {
+        id: "id",
+        name: "string",
+      });
+
+      const instance = new SortArgs();
+
+      // Test invalid sort order
+      instance.by = {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        id: "invalid" as any,
+        name: SortOrder.descending,
+      };
+      expect(instance.by).toEqual({
+        id: SortOrder.ascending,
+        name: SortOrder.descending,
+      });
+    });
+
+    it("should handle undefined sort values", () => {
+      const { SortArgs } = createEntityArgs<TestEntity>("Test", {
+        id: "id",
+        name: "string",
+      });
+
+      const instance = new SortArgs();
+
+      instance.by = {
+        id: undefined,
+        name: SortOrder.descending,
+      };
+      expect(instance.by).toEqual({
+        id: undefined,
+        name: SortOrder.descending,
+      });
+    });
+  });
+
+  describe("Type Generation", () => {
+    it("should generate unique types for different entities", () => {
```

Comment:
So you're essentially testing that the where- and sort-args are not references to the same args, right?

Reply:
Specifically whether similar entity type and args with different names results in different types. At one point, we simple had 'MetadataArgs' regardless if they where nested or partial. By now, we have `HypercertsMetadataArgs` and `CollectionMetadataArgs` (for example) and they can point to the same entity, with same args, but they are nested within different entities (Hypercerts, Collection resp.)